lisk-dex-electron/src/PlaceOrder.jsx

```diff
@@ -136,7 +136,13 @@ export default class PlaceOrder extends React.Component {
       price: null,
       amount: null,
     };
-    if (!this.state.marketMode) {
+    if (Number.isNaN(this.state.amount) || this.state.amount === '') {
+      errors.amount = 'The order amount must be a number.';
+      success = false;
+    } else if (this.state.amount > this.props.assetBalance) {
+      errors.amount = 'Insufficient balance !!';
```

Comment:
You need to account for the transaction fee as well. If your balance is 1000 and you try to send 1000, it should reject because you also need to pay 0.1 LSK transaction fee. The base transaction fee depends on the market. Look for `exchangeFeeBase` from the `/status` endpoint.

Reply:
Regarding this. shouldn't it be, the fee should be deducted from the amount being sent? I have seen same kind of behaviour on most of the exchanges. We can probably adjust for fee in estimated received value.